src/main/java/uk/gov/hmcts/ccd/data/draft/DefaultDraftGateway.java

```diff
@@ -0,0 +1,179 @@
+
+    @Inject
+    public DefaultDraftGateway(
+        final RestTemplate restTemplate,
+        final SecurityUtils securityUtils,
+        final ApplicationParams applicationParams,
+        final AppInsights appInsights) {
+        this.restTemplate = restTemplate;
+        this.securityUtils = securityUtils;
+        this.applicationParams = applicationParams;
+        this.appInsights = appInsights;
+    }
+
+    @Override
+    public Long save(final CreateCaseDraftRequest draft) {
+        try {
+            HttpHeaders headers = securityUtils.authorizationHeaders();
+            headers.add(DRAFT_ENCRYPTION_KEY_HEADER, applicationParams.getDraftEncryptionKey());
+            final HttpEntity requestEntity = new HttpEntity(draft, headers);
+            final Instant start = Instant.now();
+            HttpHeaders responseHeaders = restTemplate.exchange(applicationParams.draftBaseURL(),
+                                                                HttpMethod.POST,
+                                                                requestEntity,
+                                                                HttpEntity.class).getHeaders();
+            final Duration duration = Duration.between(start, Instant.now());
+            appInsights.trackDependency(DRAFT_STORE, "Create", duration.toMillis(), true);
+            return getDraftId(responseHeaders);
+        } catch (Exception e) {
+            LOG.warn("Error while saving draft", e);
+            throw new ServiceException(DRAFT_STORE_DOWN_ERR_MESSAGE, e);
+        }
+    }
+
+    @Override
+    public DraftResponse update(final UpdateCaseDraftRequest draft, final String draftId) {
+        try {
+            HttpHeaders headers = securityUtils.authorizationHeaders();
+            headers.add(DRAFT_ENCRYPTION_KEY_HEADER, applicationParams.getDraftEncryptionKey());
+            final HttpEntity requestEntity = new HttpEntity(draft, headers);
+            final Instant start = Instant.now();
+            restTemplate.exchange(applicationParams.draftURL(draftId), HttpMethod.PUT, requestEntity, HttpEntity.class);
+            final Duration duration = Duration.between(start, Instant.now());
+            appInsights.trackDependency(DRAFT_STORE, "Update", duration.toMillis(), true);
+            return aDraftResponse()
+                .withId(draftId)
+                .build();
+        } catch (HttpClientErrorException e) {
+            LOG.warn("Error while updating draftId={}", draftId, e);
+            if (e.getRawStatusCode() == RESOURCE_NOT_FOUND) {
+                throw new ResourceNotFoundException(String.format(RESOURCE_NOT_FOUND_MSG, draftId));
+            }
+        } catch (Exception e) {
+            LOG.warn("Error while updating draftId={}", draftId, e);
+            throw new ServiceException(DRAFT_STORE_DOWN_ERR_MESSAGE, e);
+        }
+        return null;
+    }
+
+    @Override
+    public DraftResponse get(final String draftId) {
```

Comment:
I think get methods are better return Optional in the case the entity is not found. We are using Optional in our new CaseDetailsRepository API

Reply:
It is consistent with general pattern which is handled by error handler. Both possible exceptions are converted into a generic HttpError and sent back to UI.